Modules/_sqlite/statement.c

```diff
@@ -48,13 +52,59 @@ typedef enum {
     TYPE_UNKNOWN
 } parameter_type;
 
+int pysqlite_statement_is_dml(sqlite3_stmt *st, const char *sql)
+{
+    const char* p;
+    int is_dml = 0;
+
+#ifdef HAVE_SQLITE3_STMT_READONLY
+    is_dml = ! sqlite3_stmt_readonly(st);
```

Comment:
Style nit: We don't add whitespace after `!`.

Reply:
Done.